kolibri_zim_plugin/assets/src/views/ZimContentView.vue

```diff
@@ -87,11 +98,30 @@
             elem.setAttribute('src', remapUrl);
           }
         });
-        // TODO: Wait for external resources to load, first, to prevent flash
-        //       of unstyled content.
-        this.shadow.replaceChildren(document.documentElement);
+        document.head.getElementsByTagName('link').forEach(elem => {
+          if (elem.rel === 'stylesheet' && elem.hasAttribute('href')) {
+            this.resourcesToLoad += 1;
+            elem.addEventListener('load', this.onDocumentResourceLoad, { once: true });
+            elem.addEventListener('error', this.onDocumentResourceLoad, { once: true });
+          }
+        });
+        this.loadShadow.replaceChildren(document.documentElement);
+        this.checkDocumentLoadFinished();
         return title;
       },
+      onDocumentResourceLoad() {
+        this.resourcesLoaded += 1;
+        this.checkDocumentLoadFinished();
+      },
+      checkDocumentLoadFinished() {
```

Comment:
Nit: This method is not only doing the check but also doing the replacement. So I would rename it.